Automattic/jetpack
# Protect: Handle unavailable scan status

Fixes #
In a number of scenarios where we can determine with certainty that an appropriate scan level plan is present, but the scan module is not currently available (either due to a provisioning delay or an error), we are presented with an `unavailable` status that we do not yet have handling for.

#### Changes proposed in this Pull Request:

* Adds a `setScanIsUnavailable` action
* Adds a `checkStatus` action
* Updates the `refreshStatus` action to use `checkStatus` where applicable
* Updates the `AdminPage` component to react accordingly

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203341574761581

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






* Checkout this branch
* Manually set `$status->status` to `unavailable` by adding the following to line 69-70 in `src/class-status.php` (ensure `jetpack watch plugins/protect` is running):
```
self::$status->status = 'unavailable';
self::$status->last_checked = null;
```
* Fire up Jurassic Tube and ensure you have a scan-level upgrade
* Activate Protect and proceed to the admin page
* Verify that you initially see the in-progress screen and after 10-15s the following error screen is displayed:
<img width="1214" alt="Screen Shot 2022-11-15 at 09 16 20" src="https://user-images.githubusercontent.com/43220201/202029998-948296e5-e160-40da-9bf5-79e1a5909a9d.png">

* Open your browser Developer Tools and proceed to the Network tab
* Refresh the page, and verify that an initial `status?hard_refresh=true` request is sent, followed by an additional three attempts while the in-progress screen is displayed before eventually resolving the error screen 
* Remove the lines added to `src/class-status.php`, ensure `watch` runs, and refresh the page
* Ensure the expected content display and there are no regressions in app functionality
